pkg/cmd/operator/serveprobes/scylladbapistatus.go

```diff
@@ -0,0 +1,179 @@
+}
+
+func NewScyllaDBAPIStatusCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewScyllaDBAPIStatusOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "scylladb-api-status",
+		Short: "Serves probes based on ScyllaDB API status.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(streams, cmd)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *ScyllaDBAPIStatusOptions) Validate(args []string) error {
+	var errs []error
+
+	errs = append(errs, o.ProbeServerOptions.Validate(args))
+	errs = append(errs, o.ClientConfig.Validate())
+	errs = append(errs, o.InClusterReflection.Validate())
+
+	if len(o.ServiceName) == 0 {
+		errs = append(errs, fmt.Errorf("service-name can't be empty"))
+	} else {
+		serviceNameValidationErrs := apimachineryvalidation.NameIsDNS1035Label(o.ServiceName, false)
+		if len(serviceNameValidationErrs) != 0 {
+			errs = append(errs, fmt.Errorf("invalid service name %q: %v", o.ServiceName, serviceNameValidationErrs))
+		}
+	}
+
+	return apierrors.NewAggregate(errs)
+}
+
+func (o *ScyllaDBAPIStatusOptions) Complete(args []string) error {
+	err := o.ProbeServerOptions.Complete(args)
+	if err != nil {
+		return err
+	}
+
+	err = o.ClientConfig.Complete()
+	if err != nil {
+		return err
+	}
+
+	err = o.InClusterReflection.Complete()
+	if err != nil {
+		return err
+	}
+
+	o.kubeClient, err = kubernetes.NewForConfig(o.ProtoConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes clientset: %w", err)
+	}
+
+	return nil
+}
+
+func (o *ScyllaDBAPIStatusOptions) Run(originalStreams genericclioptions.IOStreams, cmd *cobra.Command) (returnErr error) {
+	klog.Infof("%s version %s", cmd.Name(), version.Get())
+	cliflag.PrintFlags(cmd.Flags())
+
+	stopCh := signals.StopChannel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+
+	return o.RunCtx(ctx, originalStreams, cmd)
+}
+
+func (o *ScyllaDBAPIStatusOptions) RunCtx(ctx context.Context, originalStreams genericclioptions.IOStreams, cmd *cobra.Command) (returnErr error) {
+	singleServiceKubeInformers := informers.NewSharedInformerFactoryWithOptions(
+		o.kubeClient,
+		12*time.Hour,
+		informers.WithNamespace(o.Namespace),
+		informers.WithTweakListOptions(
+			func(options *metav1.ListOptions) {
+				options.FieldSelector = fields.OneTermEqualSelector("metadata.name", o.ServiceName).String()
+			},
+		),
+	)
+	singleServiceInformer := singleServiceKubeInformers.Core().V1().Services()
+
+	prober := sidecar.NewProber(
+		o.Namespace,
+		o.ServiceName,
+		singleServiceInformer.Lister(),
+	)
```

Comment:
why does this still live in sidecar?

Reply:
if it's just a matter of moving the code to a different package, and doesn't require any additional tweaks, I'd move it to the probeserver package

Reply:
> I'd move it to *the* probeserver package

just to clarify: *the* probeserver package is `/pkg/cmd/probeserver` in which it shouldn't be, so we need a new package in `/pkg`

Reply:
done